### ISSUE TITLE
Config template for RabbitMQ generates duplicate variable

### ISSUE THREAD

**Issue by jgeorge300** [pivotal-cf/om#484]
Running
`om config-template --pivnet-product-slug p-rabbitmq --product-version '1.18.4'`

generates a duplicate variable name (on-demand-broker_max_in_flight) for two different variables in resource-vars.yml and default-vars.yml The default values are different and will throw an error when trying to configure product.

**Comment by kcboyle** [comment]
Did this end up being a non-issue @jgeorge300 ?

**Comment by jgeorge300** [comment]
I was trying to re-write/delete the issue. But it is still an issue. Re-opening.

**Comment by kcboyle** [comment]
Hi @jgeorge300! Thanks for reopening. Can you give some examples of what the files generate for you, and what causes the error? If you remove the duplicate value, does the rest of the configuration work fine for configuring the tile?

**Comment by jgeorge300** [comment]
Running: 
` om config-template --output-directory templates/ --pivnet-product-slug p-rabbitmq --product-version '1.18.4' --pivnet-api-token ${PIVNET} --file-glob '*.pivotal'`
generates the following:

default-vars.yml:
`
on-demand-broker_global_service_instance_quota: 20
on-demand-broker_max_in_flight: 3
on_demand_broker_plan_1_cf_service_access: enable
`

resource-vars.yml:
`
on-demand-broker_instance_type: automatic
on-demand-broker_instances: automatic
on-demand-broker_max_in_flight: default
on-demand-broker_persistent_disk_size: automatic
`

Notice the different values for on-demand-broker_instances. To get around this, I commented out the variable in defautl-vars.yml and replaced the first reference in product.yml to the default value.

**Comment by kcboyle** [comment]
Adding PR welcome label. config-template is still marked EXPERIMENTAL, and might not get worked on right away. However, feel free to explore the codebase, test, and submit a PR to help fix the issue if you want/need it sooner!

Thanks for the detailed steps and outputs!

**Comment by kcboyle** [comment]
when providing only required properties, the following output is observed:
```
om -k --target domain.com --username domain-username --password domain-password configure-product -c /tmp/rabbit/p-rabbitmq/1.18.4-build.83/product.yml -l /tmp/rabbit/p-rabbitmq/1.18.4-build.83/default-vars.yml -l /tmp/rabbit/p-rabbitmq/1.18.4-build.83/required-vars.yml -l /tmp/rabbit/p-rabbitmq/1.18.4-build.83/resource-vars.yml -l /tmp/rabbit/p-rabbitmq/1.18.4-build.83/errand-vars.yml
configuring p-rabbitmq...
setting up network
finished setting up network
setting properties
could not execute "configure-product": failed to configure product: request failed: unexpected response from /api/v0/staged/products/p-rabbitmq-a2065a050c7b6954ba4b/properties:
HTTP/1.1 500 Internal Server Error
Transfer-Encoding: chunked
Cache-Control: no-cache, no-store
Connection: keep-alive
Content-Type: application/json; charset=utf-8
Date: Fri, 17 Apr 2020 17:32:06 GMT
Expires: Fri, 01 Jan 1990 00:00:00 GMT
Pragma: no-cache
Referrer-Policy: strict-origin-when-cross-origin
Server: Ops Manager
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Content-Type-Options: nosniff
X-Download-Options: noopen
X-Frame-Options: SAMEORIGIN
X-Permitted-Cross-Domain-Policies: none
X-Request-Id: dddab8ec-e0b0-4020-b685-b5eed6c4e49b
X-Runtime: 0.559372
X-Xss-Protection: 1; mode=block

95
{"errors":{".on-demand-broker.max_in_flight":["Value is not a number"],"base":["undefined method `map' for \"pks-az1\":String\nDid you mean?  tap"]}}
```

**Comment by kcboyle** [comment]
this will be included in om 4.7.0
